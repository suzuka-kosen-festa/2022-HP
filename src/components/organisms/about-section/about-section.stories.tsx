import type { ComponentStoryObj, ComponentMeta } from "@storybook/react";
import { AboutSection } from ".";

type T = typeof AboutSection;
type Story = ComponentStoryObj<T>;

const data = {
  children: `鈴鹿高専については本稿ホームページをご覧ください。`,
  link: "#",
  text: "鈴鹿高専ホームページ",
  title: "鈴鹿高専について",
};

export default {
  args: { children: data.children, link: data.link, text: data.text, title: data.title },
  component: AboutSection,
} as ComponentMeta<T>;

export const Default: Story = {};
