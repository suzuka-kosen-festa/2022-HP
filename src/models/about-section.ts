import type { ComponentProps, ReactHTML } from "react";

export type AboutSectionProperties = Omit<ComponentProps<ReactHTML["section"]>, "children"> & {
  buttonText: string;
  children: string;
  link: string;
  title: string;
};
