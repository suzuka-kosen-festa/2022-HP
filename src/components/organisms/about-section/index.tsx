import React from "react";
import { AnchorButton } from "src/components/atoms/button/button.stories";
import tw from "twin.macro";
import { AboutSectionProperties } from "../../../models";
import { Button } from "../../atoms/button";
import { Heading } from "../../atoms/heading";
import { Text } from "../../atoms/text";

const AboutSectionContainer = tw.section`flex-col  space-y-[calc(100vw / 63)]`;

const AboutSection: React.FC<AboutSectionProperties> = ({ children, link, text, title }) => (
  <AboutSectionContainer aria-label={"鈴鹿高専について"}>
    <Heading>{title}</Heading>
    <Text>{children}</Text>
    <Button as="a" href={link} target={"_brank"} rel={"noopener noreferrer"} role={"link"}>
      {text}
    </Button>
  </AboutSectionContainer>
);

export { AboutSection };
