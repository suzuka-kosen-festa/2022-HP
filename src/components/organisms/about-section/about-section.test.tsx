import { composeStories } from "@storybook/testing-react";
import "@testing-library/jest-dom";
import { render } from "@testing-library/react";
import * as stories from "./about-section.stories";

const { Default } = composeStories(stories);

const regionOptions = {
  name: "鈴鹿高専について",
};

const buttonOptions = {
  name: "鈴鹿高専ホームページ",
};

describe("(components) organisms/about-section", () => {
  test("to be organisms", () => {
    const { container } = render(<Default />);
    expect(container).toBeOrganism();
  });
  test("to be [role=region]", () => {
    const { getByRole } = render(<Default />);
    expect(getByRole("region", regionOptions));
  });
  test("to be [role=button]", () => {
    const { getByRole } = render(<Default />);
    expect(getByRole("button", buttonOptions));
  });
  test("take snap shot", () => {
    const { container } = render(<Default />);
    expect(container).toMatchSnapshot();
  });
});
