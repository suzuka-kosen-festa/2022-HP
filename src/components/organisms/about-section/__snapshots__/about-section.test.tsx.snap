// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(components) organisms/about-section take snap shot 1`] = `
<div>
  <section
    aria-label="鈴鹿高専について"
    class="twin-3urxqw-AboutSectionContainer e1cvkk860"
  >
    <h1
      class="twin-f7kb4u-Heading ebaq0s80"
    >
      鈴鹿高専について
    </h1>
    <p
      class="twin-w53v7j-Text epv4mo00"
    >
      鈴鹿高専については本校ホームページをご覧ください。
    </p>
    <a
      class="twin-1fs6bgp-ButtonBox elv5i9d0"
      href="#"
      rel="noopener noreferrer"
      role="link"
      tabindex="0"
      target="_brank"
    >
      <p
        class="twin-cryf0r-Text-Button epv4mo00"
      >
        鈴鹿高専ホームページ
      </p>
    </a>
  </section>
</div>
`;
